src/asgen/backends/alpinelinux/apkpkgindex.d

```diff
@@ -0,0 +1,202 @@
+    }
+
+    private Package[] loadPackages(string suite, string section, string arch)
+    {
+        auto apkRootPath = buildPath(rootDir, suite, section, arch);
+        ArchiveDecompressor ad;
+        AlpinePackage[string] pkgsMap;
+
+        foreach (packageArchivePath; dirEntries(apkRootPath, SpanMode.shallow).filter!(
+                f => f.name.endsWith(".apk")))
+        {
+            auto fileName = packageArchivePath.baseName();
+            AlpinePackage pkg;
+            if (fileName in pkgsMap)
+            {
+                pkg = pkgsMap[fileName];
+            }
+            else
+            {
+                pkg = new AlpinePackage();
+                pkgsMap[fileName] = pkg;
+            }
+
+            ad.open(packageArchivePath);
+            auto pkgInfoData = cast(string) ad.readData(".PKGINFO");
+
+            try
+            {
+                validate(pkgInfoData);
+            }
+            catch (UTFException e)
+            {
+                logError("PKGINFO file in archive %s contained invalid UTF-8, skipping!",
+                        packageArchivePath);
+            }
+
+            pkg.filename = packageArchivePath;
+            auto lines = pkgInfoData.splitLines();
+            for (auto i = 0; i < lines.length; i++)
+            {
+                /* Skip comments */
+                if (lines[i].startsWith("#"))
+                {
+                    continue;
+                }
+
+                auto line = lines[i].strip().split("#")[0];
+
+                /*
+                    When the next line doesn't contain a "=" it's a continuation
+                    of the current line
+                */
+                int j = i + 1;
+                while (lines.length > j && !lines[j].canFind("=") && !lines[j].startsWith("#"))
+                {
+                    line ~= " " ~ lines[j].strip().split("#")[0];
+                    /* Skip the next line since it was a continuation of the previous one */
+                    i++;
+                    j++;
+                }
+
+                auto infoEntry = line.split("=");
+                immutable key = infoEntry[0].strip();
+                immutable auto value = infoEntry[1].strip();
+                switch (key)
+                {
+                case "pkgname":
+                    pkg.name = value;
+                    break;
+                case "pkgver":
+                    pkg.ver = value;
+                    break;
+                case "arch":
+                    pkg.arch = value;
+                    break;
+                case "maintainer":
+                    pkg.maintainer = value;
+                    break;
+                case "pkgdesc":
+                    setPkgDescription(pkg, value);
+                    break;
+                default:
+                    /* We don't care about other entries */
+                    break;
+                }
+            }
+            pkg.contents = ad.readContents();
```

Comment:
You potentially want to filter this to exclude the `.PKGINFO` file and the signature file (not absolutely required, but may be the nice thing to do - otherwise asgen will assume these files are actually installed with every package).

Reply:
Ah, I figured asgen would filter those out anyway, so I wouldn't have to do that myself